### PR TITLE
Upgrade all subprotocols of sort

### PR DESCRIPTION
**Pull request by richajaindce**
All subprotocols now take generic Context! Whats left
1. Add malicious tests for secureapplyinv, compose, sort (will add in next PR once #265 and #257 are approved and landed since they can conflict)
2. Validate call in sort

### CODE REVIEW

File: src/protocol/sort/reshare.rs [hunk 1]
```diff
@@ -79,26 +88,53 @@ impl<F: Field> Reshare<F> {
     }
 }
 
+/// For malicious reshare, we run semi honest reshare protocol twice, once for x and another for rx and return the results
```
Comment (benjaminsavage) [inline src/protocol/sort/reshare.rs:91]: I think that this code has already landed. Can you rebase?

Comment (richajaindce) [reply src/protocol/sort/reshare.rs:91]: Yes let me do that

File: src/protocol/sort/reshare.rs [hunk 2]
```diff
@@ -100,9 +103,6 @@ impl<F: Field> Reshare<F> for MaliciousContext<'_, F> {
         record_id: RecordId,
         to_helper: Role,
     ) -> Result<Self::Share, Error> {
-        use crate::protocol::context::SpecialAccessToMaliciousContext;
-        use crate::secret_sharing::ThisCodeIsAuthorizedToDowngradeFromMalicious;
-
```
Comment (benjaminsavage) [inline src/protocol/sort/reshare.rs:105]: I'm not sure you want to do that. @martinthomson - is there a specific reason why these use statements were formerly located here?

**Review by benjaminsavage — APPROVED** [review]
I love this change!! I can't wait to land this. Just that one question.

File: src/protocol/sort/reshare.rs (hunk 2 above)
Comment (martinthomson) [reply src/protocol/sort/reshare.rs:105]: Ah, I just wanted to ensure that these were narrowly scoped.  They really shouldn't be moved to where they went here though.

Comment (richajaindce) [reply src/protocol/sort/reshare.rs:105]: oops i thought they were misplaced.. Let me bring it back here

Comment (richajaindce) [reply src/protocol/sort/reshare.rs:105]: By mistake i git commit --amend and force pushed the changes :( but it should be in correct place now

**Pull request merged**